event/event.go

```diff
@@ -0,0 +1,211 @@
+}
+
+// MouseAxes is a mouse axis event.
+type MouseAxis struct {
+	// X position of the mouse pointer.
+	X float32
+	// Y position of the mouse pointer.
+	Y float32
+	// W is the vertical position of the mouse wheel.
+	W float32
+	// Z is the horizontal position of the mouse wheel.
+	Z float32
+	// DeltaX is the change in X since last event.
+	DeltaX float32
+	// DeltaY is the change in Y since last event.
+	DeltaY float32
+	// DeltaW is the change in W since last mouse event.
+	DeltaW float32
+	// DeltaW is the change in Z since last mouse event.
+	DeltaZ float32
+}
+
+// MouseButtonDown is a mouse button press event.
+type MouseButtonDown struct {
+	// X position of the mouse pointer.
+	X float32
+	// Y position of the mouse pointer.
+	Y float32
+	// W is the vertical position of the mouse wheel.
+	W float32
+	// Z is the horizontal position of the mouse wheel.
+	Z float32
+	// DeltaX is the change in X since last event.
+	DeltaX float32
+	// DeltaY is the change in Y since last event.
+	DeltaY float32
+	// DeltaW is the change in W since last mouse event.
+	DeltaW float32
+	// DeltaW is the change in Z since last mouse event.
+	DeltaZ float32
+	// Button that was pressed.
+	Button int
+	// Pressure applied on the mouse click.
+	// It varies between 0.0 for not pressed, and 1.0 for completely pressed.
+	Pressure float32
+}
+
+// MouseButtonDown is a mouse button Release event.
+// The data is identical to a MouseButtonDown event.
+type MouseButtonUp MouseButtonDown
+
+// MouseEnter occurs when the mouse enters the view window.
+type MouseEnter struct {
+	// X position of the mouse pointer.
+	X float32
+	// Y position of the mouse pointer.
+	Y float32
+	// W is the vertical position of the mouse wheel.
+	W float32
+	// Z is the horizontal position of the mouse wheel.
+	Z float32
+	// DeltaX is the change in X since last mouse event.
+	DeltaX float32
+	// DeltaY is the change in Y since last mouse event.
+	DeltaY float32
+	// DeltaW is the change in W since last mouse event.
+	DeltaW float32
+	// DeltaW is the change in Z last mouse event.
+	DeltaZ float32
+}
+
+// MouseLeave occurs when the mouse leaves the view window.
+// The data is identical to MouseEnter
+type MouseLeave MouseEnter
+
+// ViewUpdate occurs when the application is ready to update
+// the next frame on the view port.
+type ViewUpdate struct {
+	// No data neccesary, for now.
+}
+
+// ViewSize ocurs when the size of the application's view port changes.
+type ViewSize struct {
+	// Actual, real width of the view.
+	Width float32
+	// Actual, real height of the view.
+	Height float32
+	// X position of the view on the physical screen.
+	X float32
+	// Y position of the view on the physical screen.
+	Y float32
+}
+
+// TouchBegin occurs when a touch begins.
+type TouchBegin struct {
+	// Touch ID that caused the touch event.
+	ID int
+	// X position of the event.
+	X float32
+	// Y position of the event.
+	Y float32
+	// Change in X since last touch event.
+	DeltaX float32
```

Comment:
TouchBegin doesn't have to include DeltaX and DeltaY. Let's leave only necessary members.

Reply:
Indeed, I will do so. This does mean that TouchEnd will have the same data as TouchMoved, not as TouchBegin